### PR TITLE
fix: Adjust arrangeSize for non-Stretch alignments

### LINKED ISSUE #14373: [Android] Layout trigger arrange phase multiple time
### Current behavior

Android layout phase make panel passe more than one time in the arrange phase :

- Performance issue
- the final size change sometime between the pass causing layouting issue
- the final size pass to the arrange phase differ from the other platform causing layouting issue 

In the following example in Android the arrange go three time in the black shape returning in width size 400, then 100, then 400 
In windows it's just go one time with a value of 100

Windows : 
image

android: 
image

sample app (too big to attach)
https://we.tl/t-ujU1tMKeji

### Expected behavior

- go one time in the arrange phase
- have the same result as the other platforms

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

_No response_

### Affected platforms

Android

### IDE

Visual Studio 2022

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #14373

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/Layouter/Layouter.cs [hunk 1]
```diff
@@ -255,6 +255,17 @@ public void Arrange(Rect finalRect)
 					}
 				}
 
+				// Alignment==Stretch --> arrange at the slot size minus margins
+				// Alignment!=Stretch --> arrange at the unclippedDesiredSize
+				if (Panel.HorizontalAlignment != HorizontalAlignment.Stretch)
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/Layouter/Layouter.cs:260]: Original code from the public WinUI sources:

image

Comment (Youssef1313) [reply src/Uno.UI/UI/Xaml/Controls/Layouter/Layouter.cs:260]: We also already do that for Skia/Wasm:

https://github.com/unoplatform/uno/blob/c0862855e0ed00489665fb1e7610c46eb118b78e/src/Uno.UI/UI/Xaml/FrameworkElement.Layout.crossruntime.cs#L403-L413

File: src/Uno.UI.Tests/Windows_UI_XAML_Controls/GridTests/Given_Grid.cs
Comment (MartinZikmund) [inline src/Uno.UI.Tests/Windows_UI_XAML_Controls/GridTests/Given_Grid.cs:1]: Why was this removed 🤔?

**Review by MartinZikmund — APPROVED** [review]

File: src/Uno.UI.Tests/Windows_UI_XAML_Controls/GridTests/Given_Grid.cs
Comment (Youssef1313) [reply src/Uno.UI.Tests/Windows_UI_XAML_Controls/GridTests/Given_Grid.cs:1]: Tests are moved to runtime tests because relying on a mock layouter implementation doesn't feel right. We want to test the actual implementation on all platforms
